tensorrt_llm/_torch/attention_backend/sparse/dsa.py

```
@@ -3274,9 +3321,17 @@ def get_cache_bytes_per_token(self):
         # 1 byte/element here -- it is NOT scaled by the KV cache dtype (unlike
         # the latent above). Under FP4 the indexer data portion is halved (two
-        # E2M1 codes per byte); the scale bytes are unchanged.
+        # E2M1 codes per byte); the scale bytes are unchanged. Only
+        # full-indexer local layers own a row in the masked indexer pool, so
+        # shared layers (cross-layer indexer sharing) contribute no bytes.
         indexer_data_dim = self.index_head_dim // 2 if self.use_fp4 else self.index_head_dim
-        indexer_bytes_per_token = sum(self.num_kv_heads_per_layer) * (
+        local_mask = self.indexer_k_cache_local_layer_mask
+        if local_mask is not None:
+            num_indexer_layers = sum(kv_heads for kv_heads, has_indexer in zip(
+                self.num_kv_heads_per_layer, local_mask) if has_indexer)
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)